File: 백준/Gold/17205. 진우의 비밀번호/진우의 비밀번호.java
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());
        String password = br.readLine();
        String order = "abcdefghijklmnopqrstuvwxyz";

        long result = 0;

        for (int i = 0; i < password.length(); i++) {
            char cur = password.charAt(i);
            int idx = order.indexOf(cur);

            for (int j = 0; j < N-i; j++) {
                result += idx * (long) Math.pow(26, j);
            }

            result++;
        }
        System.out.println(result);
    }
}

File: 백준/Gold/17205. 진우의 비밀번호/진우의 비밀번호.py
# 진우의 비밀번호
import sys
input = sys.stdin.readline

N = int(input())
password = input().rstrip()
order = "abcdefghijklmnopqrstuvwxyz"

result = 0

for i in range(len(password)):
    cur = password[i]
    idx = order.find(cur)

    for j in range(N-i):
        result += idx * (26**(j))

    result += 1
    
print(result)
